### PR TITLE
adding hooks, other edits and improvements

### PR DESCRIPTION
**Pull request by nickchomey**

### CODE REVIEW

**Comment by nickchomey** [comment]
Here's the edits that I've made that are relevant to your version. It adds more flexibility with settings, hooks etc... 

Sorry about the reformatting of most/all the lines - my vscode did it all automatically. 

Please let me know if you have any questions/concerns.

File: js/pn.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            var scrollOptions = {
+                left: 0,
+                top: targetTop,
+                behavior: "smooth",
+            };
+
+            window.scrollTo(scrollOptions);
+        }
+    });
+
```
Comment (naotoonodera) [inline js/pn.js:86]: What is this for?

File: js/pn.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    pn.on("hidePopup", function () {});
+    pn.on("select", function (properties) {});
+    pn.on("selectNode", function (properties) {});
+    pn.on("selectEdge", function (properties) {});
+    pn.on("deselectNode", function (properties) {});
+    pn.on("deselectEdge", function (properties) {});
+    pn.on("hoverNode", function (properties) {});
+    pn.on("hoverEdge", function (properties) {});
+    pn.on("blurNode", function (properties) {});
+    pn.on("blurEdge", function (properties) {});
```
Comment (naotoonodera) [inline js/pn.js:105]: What are these for?

**Comment by nickchomey** [comment]
closed this to submit another. Something got corrupted with this one.

**Pull request closed without merging**